lxd/db/profiles.go

```diff
@@ -74,17 +59,39 @@ func ProfileToAPI(profile *Profile) *api.Profile {
 	}
 	p.Description = profile.Description
 	p.Config = profile.Config
-	p.Devices = profile.Devices
+	p.Devices = DevicesToAPI(profile.Devices)
 
 	return p
 }
 
 // ProfileFilter specifies potential query parameter fields.
 type ProfileFilter struct {
+	ID      *int
 	Project *string
 	Name    *string
 }
 
+// GetProfileUsedBy returns all the instances that use the given profile.
+func (c *ClusterTx) GetProfileUsedBy(profile Profile) ([]string, error) {
+	usedBy := []string{}
+
+	profileInstances, err := c.GetProfileInstances()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, instanceID := range profileInstances[profile.ID] {
+		instanceURIs, err := c.GetInstanceURIs(InstanceFilter{ID: &instanceID})
+		if err != nil {
+			return nil, err
+		}
+
+		usedBy = append(usedBy, instanceURIs...)
+	}
+
+	return usedBy, nil
+}
+
```

Comment:
@tomponline Alright, I've switched `Profiles.Instances` back into `Profiles.UsedBy` and added this manual method `GetProfileUsedBy`.

For now I've left `UsedBy` in place for `Project` and `Profile`, but I've made the `GetProfileUsedBy` and `GetProjectUsedBy` methods just return what one object is used by, so that it's easy to remove the field and get `UsedBy`s only when we want them in a different PR.

Reply:
Thanks will look at this shortly.